packages/fuels-types/src/lib.rs

```diff
@@ -52,6 +52,16 @@ pub struct ABIFunction {
     pub attributes: Option<Vec<Attribute>>,
 }
 
+impl ABIFunction {
+    pub fn is_payable(&self) -> bool {
+        if let Some(attributes) = &self.attributes {
+            return attributes.iter().any(|attr| attr.name == "payable");
+        }
+
+        false
```

Comment:
another way of writing it. I've no strong preference either way. I just wanted to point out the ability to iterate through an `Option` and flatten the contained `Vec`.
```suggestion
        self.attributes
            .iter()
            .flatten()
            .any(|attr| attr.name == "payable")
```